### PR TITLE
Correctly normalize package names. Fixes #2956 and #3002

### LINKED ISSUE #2956: Pipenv install parsing wrong package name
### Issue description
After the latest patch for the incompatibility problem with pip 18.1, the pipenv parsing of the package name during installation no longer is working (it's replacing `.` in the package name for a `-` and this leads to pip not finding the available versions).

This happens with pip 18.1 as well as 18.0.

### Expected result
`pipenv install zope.interface==4.5.0` should install from this package https://pypi.org/project/zope.interface/ 

### Actual result
Instead it translates to `zope-interface==4.5.0` and then it don't find the package:
```
Could not find a version that satisfies the requirement zope-interface==4.5.0 (from versions: 4.4.0, 4.4.1, 4.4.2, 4.4.3)
No matching distribution found for zope-interface==4.5.0
```

### Steps to replicate
create new virtualenv with latest pipenv version (2018.10.9) to see it break, and test it with  the previews one (2018.7.1) where it's still ok.

**obs**: I'm not sure if this is a bug in pip or pypi (not finding the correct version when replacing `.` with `-`), but since it started happening with the latest pipenv release then I believe it is actually related to pipenv.

### PR DESCRIPTION
**Pull request by orf**
Fixes #2956 and #3002

### CODE REVIEW

File: pipenv/patched/notpip/_internal/index.py [hunk 1]
```diff
@@ -766,7 +766,7 @@ def egg_info_matches(
         return full_match[full_match.index('-'):]
     name = match.group(0).lower()
     # To match the "safe" name that pkg_resources creates:
-    name = name.replace('_', '-')
+    name = name.replace('_', '-').replace('.', '-')
```
Comment (techalchemy) [inline pipenv/patched/notpip/_internal/index.py]: ^_^ I am actually tempted to do this, but I have a preference for not patching pip whenever possible.

I'm likely going to hold off on accepting this for the next release to see about getting feedback from the pip maintainers when I have a bit more clarity on all of the places we are currently working around this problem, which I think is pretty numerous by now...

Comment (orf) [reply pipenv/patched/notpip/_internal/index.py]: It's up to you if you accept it, and I understand if you do not, but this is currently breaking our (am I am sure others) workflow entirely. I can't see a workaround apart from downgrading

Comment (techalchemy) [reply pipenv/patched/notpip/_internal/index.py]: @orf, apart from the issues mentioned and the fixes for those specifically (which are in master), can you describe what is continuing to cause you problems? I.e. with the current master branch, how do I see a problem?

Comment (orf) [reply pipenv/patched/notpip/_internal/index.py]: I will get back to you here tomorrow morning, after I try installing the specific pipfile I have with the current master

Comment (techalchemy) [reply pipenv/patched/notpip/_internal/index.py]: ah, I see the issue in question.  This should be resolved by #3007

**Comment by cjerdonek** [comment]
@techalchemy I don't know much about this aspect of pip. Note that @uranusjr just opened this PR which touches this very code: https://github.com/pypa/pip/pull/5875
Does that PR address the issue being raised here?

File: pipenv/patched/notpip/_internal/index.py (hunk 1 above)
Comment (orf) [reply pipenv/patched/notpip/_internal/index.py]: Awesome! Thanks for your continued work on pipenv, and I'm sorry for the volume of issues!

I will confirm tomorrow that this is fixing it, or though I cannot see any code that would after my brief lookm

**Comment by techalchemy** [comment]
We don't even communicate that well when we are co-maintainers of the same projects

I don't think that one will fix this issue, but it is another case where this code is in disagreement to be sure

**Comment by cjerdonek** [comment]
Yeah, I figured that might have been happening so wanted to be sure you were at least alerted here!

**Comment by techalchemy** [comment]
@uranusjr nope, as far as I understand we don't need that change for this to work

**Comment by techalchemy** [comment]
closing for now since we will catch this upstream in the next release of pip, thanks for the PR and thanks for the input all

**Pull request closed without merging**